optimum/intel/ipex/modeling_base.py

```diff
@@ -346,15 +341,44 @@ def __init__(
             )
         except AttributeError:
             self.model_cls = get_model_class(self.config, AutoModelForCausalLM._model_mapping)
-        self._reorder_cache = self.model_cls._reorder_cache.__get__(self)
-        self.prepare_inputs_for_generation = self.model_cls.prepare_inputs_for_generation.__get__(self)
```

Comment:
I think it makes sense to keep it as `_reorder_cache` and `prepare_inputs_for_generation` can be different depending on the modeling